netket/utils/group/_permutation_group.py

```diff
@@ -66,6 +66,10 @@ def __repr__(self):
     def __array__(self, dtype: DType = None):
         return np.asarray(self.permutation, dtype)
 
+    def permute_index(self, x: Array):
+        """Returns the image of indices `x` under the permutation"""
+        return np.argsort(self.permutation)[x]
```

Comment:
I'm not a fan of this name... If I am not mistaken, `indices` are called `id/ids` in the Lattice class for example in `Lattice.id_from_position` so I'd prefer we are consistent with calling those id.

Then, `permute`, while correct is very specific... we could actually call this `apply_to_id`?

Reply:
the tests are also complaining that this function is not tested on Permutation, but only on `PermutationGroup`. 
Can you add a test for this specific function?

Reply:
> the tests are also complaining that this function is not tested on Permutation, but only on `PermutationGroup`. Can you add a test for this specific function?

Done. The test highlights that `Identity` (of course) doesn't have `apply_to_id`. Should we add an `Element` optional arg to `PermGroup.apply_to_id` as a convenience like it is here: https://github.com/netket/netket/blob/08986f3bc85cf3ddd059d21d6f540a510d4b834e/netket/utils/group/_point_group.py#L405-L413

Reply:
I don't understand what you are proposing...

Reply:
Something like
```python
def permute_index(self, x: Array, g: Optional[Element] = None):
        """Returns the image of indices `x` under the permutation"""
    if g is None:
        return np.argsort(self.permutation)[x]
    else:
        if isinstance(g, Permutation):
            return ...
        elif isinstance(g, Identity):
            return ...
        else:
            raise ....
```
The point is that this way the identity element can be treated the same way as permutations.

Reply:
I've just noticed this comment is on `Permutation.apply_to_id`. I meant `PemrutationGroup.apply_to_id`